lib/core/msgchan/msgchan_priv.h

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) <2022>, Intel Corporation
+ */
+
+#ifndef _MSGCHAN_PRIV_H_
+#define _MSGCHAN_PRIV_H_
+
+#include <sys/queue.h>
+#include <cne_common.h>
+#include <cne_ring.h>
+#include <cne_ring_api.h>
+
+/**
+ * @file
+ * Message Channels
+ *
+ * Create a message channel using two lockless rings to communicate between two threads.
+ *
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define MC_COOKIE    ('C' << 24 | 'h' << 16 | 'a' << 8 | 'n')
+#define MC_NAME_SIZE (CNE_NAME_LEN + 4)
+enum { MC_RECV_RING = 0, MC_SEND_RING };
+
+typedef struct msg_chan {
+    TAILQ_ENTRY(msg_chan) next;      /**< Next entry in the global list. */
+    struct msg_chan *parent;         /**< Pointer to parent channel. */
+    char name[MC_NAME_SIZE];         /**< The name of the message channel */
+    uint32_t cookie;                 /**< Cookie value to test for valid entry */
+    bool mutex_inited;               /**< Flag to detect mutex is inited */
+    pthread_mutex_t mutex;           /**< Mutex to protect the attached list */
+    cne_ring_t *rings[2];            /**< Pointers to the send/recv rings */
+    int child_count;                 /**< Number of children */
+    TAILQ_HEAD(, msg_chan) children; /**< List of attached children */
+    uint64_t send_calls;             /**< Number of send calls */
+    uint64_t send_cnt;               /**< Number of objects sent */
+    uint64_t recv_calls;             /**< Number of receive calls */
+    uint64_t recv_cnt;               /**< Number of objects received */
+    uint64_t recv_timeouts;          /**< Number of receive timeouts */
+} msg_chan_t;
+
+#define MC_CREATE_RECURSIVE_MUTEX(mutex)                                      \
```

Comment:
Can this be a functino? Doesn't seem like it needs to be a macro to me. Also, the caller might wnat to know if any of these functinos fail as it should'nt continue processing.

Reply:
Create a function